### PR TITLE
fix: sanitize html in highlighted code

### PR DESCRIPTION
**Pull request by CrabNejonas**
This adds `sanitize-html` to sanitize the HTML we include in the page through `innerHTML` which can *technically* be used to execute untested code (the chance is very low, but sanitizing here is low overhead afaik and simple to include)

resolves DT-26

### CODE REVIEW

**Review by lucasfernog-crabnebula — APPROVED** [review]

File: web-client/src/components/span/span-detail.tsx [hunk 1]
```diff
@@ -70,15 +71,23 @@ export function SpanDetail(props: { span: UiSpan }) {
   const [codeHtml] = createResource(
     () => [code(), createHighlighter()] as const,
     async ([code, highlighter]) => {
-      return code === null
-        ? null
```
Comment (lucasfernog-crabnebula) [inline web-client/src/components/span/span-detail.tsx:74]: we're not returning `null` anymore, i hope this doesn't break anything

**Comment by johann-crabnebula** [comment]
I would like to split the shiki highlighting and and the source loading. So that we can simply give a string value to the shiki highlight component. That would reduce redundancy.

**Review by johann-crabnebula — CHANGES_REQUESTED** [review]
I will open a PR with a little proposal.

**Comment by johann-crabnebula** [comment]
Here is the PR: https://github.com/crabnebula-dev/devtools/pull/166
@CrabNejonas 

Let me know what you think.

**Review by atila-crabnebula — CHANGES_REQUESTED** [review]
It seems there's [no consensus](https://github.com/crabnebula-dev/devtools/pull/166#discussion_r1426125064) about which sanitization library is better to use.

In addition, we have talked about moving code parsing and highlighting out of the main thread **before** doing this and replacing Shiki with Shikiji before.

So let's hold out on merging temporary code to `main`.

**Pull request closed without merging**